system/jlib/jtrace.cpp

```diff
@@ -734,6 +742,44 @@ class CSpan : public CInterfaceOf<ISpan>
         return span ? span->IsRecording() : false;
     }
 
+    virtual void setSpanStatus(bool spanFailed, const char * statusMessage)
+    {
+        if (span != nullptr)
+        {
+            span->SetStatus(spanFailed ? opentelemetry::trace::StatusCode::kError :  opentelemetry::trace::StatusCode::kOk, statusMessage);
+        }
+    }
+
+    virtual void recordError(const SpanError error, bool spanFailed)
+    {
+        if (span != nullptr)
+        {
+            if (spanFailed)
+                span->SetStatus(opentelemetry::trace::StatusCode::kError, error.errorMessage);
+
+            //https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/
+            //The event name MUST be "exception".
+            //The table below indicates which attributes should be added to the Event and their types.
+            //exception.escaped	boolean	SHOULD be set to true if the exception event is recorded at a point where it is known that the exception is escaping the scope of the span. [1]		Recommended
```

Comment:
Should there be boolean parameters for
escapesSpan/terminatesSpan
recordStackTrace
?

Reply:
Up for discussion, I honestly feel you would have requested to remove them had I introduced them and opted to include this comment instead. Please just let me know.

Reply:
I don't have strong opinions either way.
* Stack traces could be added later, and would probably be better controlled by a thread-local flag to trace exceptions.
 * The escaped flag would need to be passed in by the caller.  If we want to support it then now is probably the time to add it.